code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,558 @@
+
+	animate(ghost.get_filter("wraith_hyperposition_windup_filter_1"), x = 30*rand() - 15, y = 30*rand() - 15, time = 0.5 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+	animate(ghost.get_filter("wraith_hyperposition_windup_filter_2"), x = 30*rand() - 15, y = 30*rand() - 15, time = 0.5 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+
+	if(!do_after(ghost, hyperposition_windup, TRUE, ghost, BUSY_ICON_FRIENDLY)) //Channel time/wind up
+		ghost.visible_message("<span class='xenowarning'>The space around [ghost] abruptly stops shifting and wavering.</span>", \
+		"<span class='xenodanger'>We abort teleporting back to our warp shadow.</span>")
+		ghost.remove_filter("wraith_hyperposition_windup_filter_1")
+		ghost.remove_filter("wraith_hyperposition_windup_filter_2")
+		return fail_activate()
+
+	ghost.remove_filter("wraith_hyperposition_windup_filter_1")
+	ghost.remove_filter("wraith_hyperposition_windup_filter_2")
+
+	if(!can_use_action()) //Check if we can actually position swap again due to the wind up delay.
+		return fail_activate()
+
+	var/beacon_turf = get_turf(shadow)
+
+	shadow.forceMove(get_turf(ghost)) //Move the beacon to where we are leaving
+	shadow.dir = ghost.dir //Have it imitate our facing
+	shadow.pixel_x = ghost.pixel_x //Inherit pixel offsets
+	teleport_debuff_aoe(ghost) //Apply tele debuff
+
+	ghost.forceMove(beacon_turf) //Move to where the beacon was
+	teleport_debuff_aoe(ghost) //Apply tele debuff
+
+	ghost.visible_message("<span class='warning'>\ [ghost] suddenly vanishes in a vortex of warped space!</span>", \
+	"<span class='xenodanger'>We teleport, swapping positions with our warp shadow. Our warp shadow has moved to [get_area(shadow)] (X: [shadow.x], Y: [shadow.y]).</span>", null, 5) //Let user know the new location
+
+	GLOB.round_statistics.wraith_hyperpositions++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_hyperpositions") //Statistics
+	succeed_activate()
+	add_cooldown()
+
+
+/datum/action/xeno_action/hyperposition/on_cooldown_finish()
+	to_chat(owner, "<span class='xenonotice'>We are able to swap locations with our warp shadow once again.</span>")
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+// ***************************************
+// *********** Phase Shift
+// ***************************************
+/datum/action/xeno_action/phase_shift
+	name = "Phase Shift"
+	action_icon_state = "phase_shift"
+	mechanics_text = "We force ourselves temporarily out of sync with reality, allowing us to become incorporeal and move through any physical obstacles for a short duration."
+	plasma_cost = 75
+	cooldown_timer = 20 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_PHASE_SHIFT
+	var/turf/starting_turf = null
+	var/phase_shift_active = FALSE
+
+
+/datum/action/xeno_action/phase_shift/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/wraith/ghost = owner
+	var/atom/movable/ghost_movable = owner
```

Comment:
/mob/living/carbon/xenomorph/wraith/ghost is also an atom/movable, no need for the ghost_movable var

Reply:
this

Reply:
still standing